### PR TITLE
Update missed references in Delivery RFC

### PR DESCRIPTION
**Pull request by sclevine**
Signed-off-by: Stephen Levine <slevine@vmware.com>

### CODE REVIEW

**Comment by sclevine** [comment]
Note: we should decide if it makes sense to update RFCs once they are merged. Ideally, future changes are captured in documentation/specification, and not by updating past proposals.

**Review by waciumawanjohi — APPROVED** [review]

**Pull request merged**
